### ISSUE TITLE
[question]: Does MemLab supports Typescript and EMS?

### ISSUE THREAD

**Issue by Gumichocopengin8** [facebookincubator/memlab#34]
Thank you for making a great framework.
Does MemLab supports Typescript and EMS?

**Comment by JacksonGL** [comment]
Yes, MemLab supports TypeScript. The MemLab npm packages export types, here is [an example](https://facebook.github.io/memlab/docs/api/heap-analysis/src/variables/takeNodeFullHeap/) and [another example](https://facebook.github.io/memlab/docs/api/core/src/functions/tagObject/). But remember to compile your code before passing it as flags (e.g., `--scenario`, `--leak-filter`) to MemLab or before running your TypeScript code that uses the MemLab APIs.

**Comment by gyzerok** [comment]
@JacksonGL hello! I am trying to write scenario file in typescript, however `memlab` package does not seem to export `IScenario` interface. What am I missing?

<img width="840" height="244" alt="Image" src="https://github.com/user-attachments/assets/16adfc17-f797-4cd5-a629-095170512520" />

**Comment by JacksonGL** [comment]
@gyzerok memlab exports its types. You can import the `IScenario` interface using:

```typescript
import type {IScenario} from '@memlab/core';
```

A full list of exported interfaces and APIs is available in the documentation:
[https://facebook.github.io/memlab/docs/api/core/src/](https://facebook.github.io/memlab/docs/api/core/src/)

**Comment by gyzerok** [comment]
> [@gyzerok](https://github.com/gyzerok) memlab exports its types. You can import the `IScenario` interface using:
> 
> import type {IScenario} from '@memlab/core';
> 
> A full list of exported interfaces and APIs is available in the documentation: https://facebook.github.io/memlab/docs/api/core/src/

Yeah, but memlab specifies that one as a `dependency`, not a `peerDependency`, so I need to install it separately and it might cause version incompatibilty. Is there a reason why `config` is reexported in the main package but not the types?

**Comment by JacksonGL** [comment]
This is not an intentional design decision. We can re-export those types in `memlab`, so you can use them like this:

```typescript
import type {IScenario, IHeapSnapshot, IHeapNode} from 'memlab';
```

Would that work for your use case?

**Comment by gyzerok** [comment]
> This is not an intentional design decision. We can re-export those types in `memlab`, so you can use them like this:
> 
> ```typescript
> import type {IScenario, IHeapSnapshot, IHeapNode} from 'memlab';
> ```
> 
> Would that work for your use case?

Yeah, thats exactly how I've expected it to work. Thank you!
